### PR TITLE
Use a single `MetaGraph` that combines `graph_flow` and `graph_control`

### PR DESCRIPTION
**Pull request by SouthEndMusic**
Fixes https://github.com/Deltares/Ribasim/issues/784.

### CODE REVIEW

**Comment by SouthEndMusic** [comment]
Locallly a test from `Expand logic_mapping` fails, but not when run in the REPL 🤔

File: core/src/allocation.jl [hunk 1]
```diff
@@ -1,30 +1,47 @@
+"""
+Temporary solution before 'edge_ids_flow_inv' is no longer needed.
```
Comment (SouthEndMusic) [inline core/src/allocation.jl:2]: This is a temporary fix, will be removed after refactoring allocation with the metagraph

File: core/src/utils.jl [hunk 2]
```diff
@@ -12,25 +12,73 @@ function valid_edge_types(db::DB)::Bool
...
+            error("Invalid edge type $edge_type.")
         end
+        graph[NodeID(from_node_id), NodeID(to_node_id)] =
+            EdgeMetadata(EdgeID(fid), edge_type)
     end
-    return graph, edge_ids, edge_connection_types
+    return graph
+end
+
+function inneighbor_labels_type(
```
Comment (SouthEndMusic) [inline core/src/utils.jl:45]: This and the following 2 functions might be badly named. Also, they might be allocating and thus slow.

File: core/src/utils.jl [hunk 3]
```diff
@@ -77,8 +125,8 @@ function get_storage_from_level(basin::Basin, state_idx::Int, level::Float64)::F
     bottom = first(level_discrete)
 
     if level < bottom
-        node_id = basin.node_id[state_idx]
-        @error "The level $level of basin #$node_id is lower than the bottom of this basin $bottom."
+        node_id = basin.node_id[NodeID(state_idx)]
```
Comment (SouthEndMusic) [inline core/src/utils.jl:128]: This seems like an ugly fix

**Comment by visr** [comment]
Ha, I messed up here, and reviewed #807 instead. I was wondering why it was so big haha. Though given that, maybe we should just merge #807 when the comments are addressed and close this.

**Comment by visr** [comment]
Closing in favor of #807.

**Pull request closed without merging**